### PR TITLE
add migration automagic and detection for unmigrated data

### PR DESCRIPTION
**Pull request by technocake**
Checks if any jfile in .goto/projects
If so, prompts user to migrate data

Added end_to_end tests

### CODE REVIEW

**Comment by Arxcis** [comment]
Issue after `git pull && make install` latest `develop` branch
image

```
jonasjso in ~/git/github/technocake/goto on develop
$ goto list
Traceback (most recent call last):
  File "/usr/local/bin/the_real_goto.py", line 11, in <module>
    load_entry_point('magicgoto', 'console_scripts', 'the_real_goto.py')()
  File "/Users/jonasjso/git/github/technocake/goto/goto/the_real_goto.py", line 62, in main
    magic = GotoMagic(project)
  File "/Users/jonasjso/git/github/technocake/goto/goto/gotomagic/magic.py", line 38, in __init__
    utils.create_project_folder(project, scope, GOTOPATH)
  File "/Users/jonasjso/git/github/technocake/goto/goto/gotomagic/utils.py", line 41, in create_project_folder
    os.makedirs(project_folder)
  File "/usr/local/Cellar/python@2/2.7.16/Frameworks/Python.framework/Versions/2.7/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 20] Not a directory: '/Users/jonasjso/.goto/projects/goto/private'
```

File: goto/gotomagic/utils.py [hunk 1]
```diff
@@ -68,22 +76,84 @@ def detect_unescaped_ampersand_url():
...
+    '''
+    if sys.version_info[0] == 2:
+        if not isinstance(message, unicode):  # noqa
+            message = unicode(message, 'utf-8')  # noqa
+        message = message.encode('utf-8')
+
+    print(message)
+
+
+def fix_python2():
```
Comment (Arxcis) [inline goto/gotomagic/utils.py:161]: Ambitious

File: goto/gotomagic/utils.py [hunk 2]
```diff
@@ -68,22 +76,84 @@ def detect_unescaped_ampersand_url():
...
+    '''
+        Assume all input is utf-8.
+        I am sure this will cause issues
+    '''
+    if sys.version_info[0] == 2:
+        sys.argv = map(lambda arg: unicode(arg, 'utf8'), sys.argv)  # noqa
```
Comment (Arxcis) [inline goto/gotomagic/utils.py:167]: This line is beautifully powerful

**Review by Arxcis — COMMENTED** [review]
We need a fix to the bug described in a comment. Is it within the scope of this PR?

**Review by technocake — COMMENTED** [review]
Much clean

**Review by Arxcis — APPROVED** [review]

**Pull request merged**